descartes_light/bgl/include/descartes_light/bgl/impl/utils.hpp

```diff
@@ -68,18 +68,21 @@ SubGraph<FloatType> createDecoratedSubGraph(const BGLGraph<FloatType>& g)
 
   EdgeIt<FloatType> first, last;
   boost::tie(first, last) = boost::edges(g);
-  for (auto it = first; it != last; ++it)
+  if (first != last) // Fix a clang tidy warning
```

Comment:
FWIW, you can also do `for (auto it = first; it < last; ++it)`

Reply:
Actually is the comment correct? It seems like you added a check to make sure the for loop is valid, not fixed a clang-tidy warning. Also I think the if-statement is unnecessary, since the for loop won't run if `first == last`. I would suggest dropping this change

Reply:
That doesn't work. It says `invalid operands to binary expression`

Reply:
Okay. I think that may just be for `std::iterator` and not a BGL vertex. In either case, is the if-statement necessary?

Reply:
It is for it to pass clang-tidy. My machine was complaining about a `-Werror=maybe-uninitialized`. Which I don't really see how it could be uninitialized, but I guess somehow it's possible.

Reply:
Apparently clang-tidy only complains about it if you are building in release and our CI is building in debug so I guess I'll revert this change. That is super weird.